### DISCUSSION TITLE
How to get the state/context of independent machines?

### DISCUSSION THREAD

**Discussion by joelrozen** [statelyai/xstate#3731] [Q&A] [answered]
I've created an "AutoSaveInput" machine. Essentially it can be "idle", "saving", "saved" or "error".

![image](https://user-images.githubusercontent.com/20792877/210039676-bd7b91be-cfb5-4707-a731-55046f785525.png)

When it blurs, it invokes a promise that makes a fetch request with the id of the input and value of the target. 

The way I use the machine is in a component called `<AutoSaveInput>`  which takes things like name, id, label, required, endpoint etc etc, the usual stuff.

Then in my form I just have a bunch of <AutoSaveInput {...stuff} /> throughout. 

This is a multi-step form (similar to an onboarding flow), so once all the fields on the first "page" are complete, I want them to be able to click a "Next" button. However I want the "Next" button to be disabled until *every* AutoSaveInput is in a "saved" state. 

How can I achieve this without using parent machine? Parent machine won't work because the fields come dynamically from a server, so sometimes the form has 2 inputs, sometimes 20, so I can't hard code all the possible fields as child states of the parent.

Is what I'm trying to achieve possible with subscriptions? I tried to play around with subscribing to the service in each AutoSaveInput but then couldn't work out how to get that subscription back up to the parent <Form>

Appreciate suggestions!

**Comment by davidkpiano** [comment, marked as answer]
Do you have an array of these autosave inputs? 

Seems like there should be some "parent" (maybe not a machine, up to you) that keeps track of all the autosave input actors. Then, you can check them all via `inputs.every(input => input.getSnapshot().matches('saved'))`.

Instead of invoking each machine via `useMachine` in each component, you can invoke them at some higher level:

```js
const inputs = data.fields.forEach(field => {
  return interpret(autoSaveInputMachine.withContext(...)).start();
});
```

**Reply by joelrozen** [reply]
Ah, this is perfect! In my head I had it that interpret() was used to make a singleton actor so if I passed in the same machine it would return the same actor, it just clicked that the example in the docs does that because you are exporting the actor to reuse not because of interpret() itself 😅
